src/sunmatrix/dense/sunmatrix_dense.c

```diff
@@ -333,11 +326,6 @@ int SUNMatSpace_Dense(SUNMatrix A, long int* lenrw, long int* leniw)
 
 static booleantype compatibleMatrices(SUNMatrix A, SUNMatrix B)
 {
-  /* both matrices must be SUNMATRIX_DENSE */
```

Comment:
I'm unclear on why this check was removed.  Is the idea that if one/both are not `SUNMATRIX_DENSE` then the subsequent accessors `SM_ROWS_D()` and `SM_COLUMNS_D()` will fail, and thus the following call will `return SUNFALSE` anyways?